Prox/Prox/Models/Place.swift

```diff
@@ -237,48 +255,95 @@ struct OpenHours {
                 return nil
             }
 
-            guard let openTime = getDate(fromServerStr: lastInterval[0]),
-                    var closeTime = getDate(fromServerStr: lastInterval[1]) else {
+            guard let openTime = getTimeComponents(fromServerStr: lastInterval[0]),
+                    let closeTime = getTimeComponents(fromServerStr: lastInterval[1]) else {
                 print("lol unable to convert date str, \(lastInterval[0]) & \(lastInterval[1]), to Date")
                 return nil
             }
 
-            if closeTime < openTime { // i.e. open overnight.
-                closeTime.addTimeInterval(TimeInDay) // Date defaults to today – this happens tomorrow.
-            }
-
-            out[day] = (open: openTime, close: closeTime)
+            out[day] = (openTime: openTime, closeTime: closeTime)
         }
 
         return OpenHours(hours: out)
     }
 
-    private static func getDate(fromServerStr serverStr: String) -> Date? {
-        guard serverStr.characters.count != 4 || serverStr.characters.count != 5 else { // "1:00" or "10:00"
-            return nil // the calling function logs.
+    private static func getTimeComponents(fromServerStr serverStr: String) -> DateComponents? {
+        let timeComponents = serverStr.components(separatedBy: ":")
+        guard timeComponents.count == 2,
+            let hours = Int(timeComponents[0]), let minutes = Int(timeComponents[1]) else {
+            return nil
+        }
+
+        let dateComponents = DateComponents(calendar: OpenHours.calendar, timeZone: nil, era: nil, year: nil, month: nil, day: nil, hour: hours, minute: minutes, second: nil, nanosecond: nil, weekday: nil, weekdayOrdinal: nil, quarter: nil, weekOfMonth: nil, weekOfYear: nil, yearForWeekOfYear: nil)
+
+        return dateComponents
+    }
+
+    func isOpen(atTime time: Date) -> Bool {
+        let dayOfWeek = DayOfWeek.forDate(time)
+        print("hours \(hours)")
+        print("The day of the week is \(dayOfWeek)")
+        guard let openTime = hours[dayOfWeek]?.openTime,
+            let openingTime = date(forTime: openTime, onDate: time),
+            let closeTime = hours[dayOfWeek]?.closeTime,
+            var closingTime = date(forTime: closeTime, onDate: time) else {
+            return false
         }
 
-        return inputFormatter.date(from: serverStr)
+        if time < openingTime {
+            print("The previous day of the week is \(dayOfWeek.previousWeekday())")
+            if let yesterdaysClosingHours = hours[dayOfWeek.previousWeekday()]?.closeTime,
+                let yesterdaysClosingTime = date(forTime: yesterdaysClosingHours, onDate: time) {
+                return time < yesterdaysClosingTime
```

Comment:
This only applies if yesterday's closing time < yesterday's opening time.

For example, if we closed yesterday at 11:59pm, this would almost always be true.